cmd/config.go

```diff
@@ -254,9 +259,78 @@ func validateConfig(conf Config, isExecutable func(string) bool) error {
 		}
 	}
 
+	// If there are thresholds to validate, the registry paramater is not allowed to be nil.
+	// Note that the reason for passing it as a pointer in the first place is
+	// because it holds a Mutex, which effectively forbids passing it by value.
+	if conf.Thresholds != nil && len(conf.Thresholds) > 0 && registry == nil {
+		err := fmt.Errorf(
+			"unable to validate thresholds configuration; " +
+				"reason: provided registry is nil",
+		)
+		errList = append(errList, err)
+		return consolidateErrorMessage(errList, "there were problems while validating the specified script configuration: ")
+	}
```

Comment:
nit: this seems like defensive programming to me - there is no case where `registry` should be `nil` to begin with IMO.

Reply:
I agree, this is defensive programming; however, I would like to keep it that way. I'm happy to discuss how we could improve this, and it's probably my C programming habits kicking in, but the pointer should be checked. It's defensive in the sense that I don't want to risk a potential segfault, or security issue to go through to production because my future self did something stupid and somehow ended up passing nil in there :)